cassiopeia/solver/solver_utilities.py

```diff
@@ -228,3 +228,25 @@ def _to_newick_str(g, node):
 def post_process_tree(T, cm):
     # raise NotImplementedError()
     pass
+
+
+def negative_log_prob_weights(priors):
+    """Generates a dictionary of negative log probabilities from priors.
+
+    Generates a dicitonary of weights for use in algorithms that inheret the
+    GreedySolver from given priors.
+
+    Args:
+        priors: A dictionary of prior probabilities for each character/state
+            pair
+
+    Returns:
+        A dictionary of weights for each character/state pair
+    """
+    weights = {}
+    for character in priors:
+        state_weights = {}
+        for state in priors[character]:
+            state_weights[state] = -np.log(priors[character][state])
+        weights[character] = state_weights
+    return weights
```

Comment:
Do you think it would be worth generalizing this to something like `transform_priors` that takes in a function to apply to all priors?

Reply:
yes I do think that is a good idea

Reply:
Sweet, let's make this change and then we can merge this code in.